### DISCUSSION TITLE
Does fastify/ajv automatically compile schema ?

### DISCUSSION THREAD

**Discussion by DarkGL** [fastify/fastify#5578] [Q&A] [answered]
Does fastify/ajv automatically compile schema or you need to use setValidatorCompiler to force it ?

[https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/)

**Comment by Eomm** [comment, marked as answer]
Fastify compiles it automatically using ajv.

The `setValidatorCompiler` is mostly used to replace the Fastify's default ajv module and replace it with another module such [joi](https://github.com/Eomm/joi-compiler).
